vmm-reference-boot.md

```diff
@@ -0,0 +1,125 @@
+```
+Header field: `kernel_alignment`. Alignment unit required by a relocatable kernel.
+
+```rs
+const EBDA_START: u64 = 0x0009_fc00;
+```
+Start address for the EBDA (Extended Bios Data Area). Older computers (like the one this VMM emulates) typically use 1 KiB for the EBDA, starting at 0x9fc00.
+
+* MEMORY LAYOUT
+
+The traditional memory map for the kernel loader, used for Image or
+zImage kernels, typically looks like:
+
+<pre>
+	    |			            |                                           
+0A0000	+------------------------+
+	    |  Reserved for BIOS	|	Do not use.  Reserved for BIOS EBDA.    
+09A000	+------------------------+                                          
+	    |  Command line		    |                                           
+	    |  Stack/heap		    |	For use by the kernel real-mode code.   
+098000	+------------------------+	                                        
+	    |  Kernel setup		    |	The kernel real-mode code.              
+090200	+------------------------+                                          
+	    |  Kernel boot sector	|	The kernel legacy boot sector.          
+090000	+------------------------+                                          
+	    |  Protected-mode kernel|	The bulk of the kernel image.           
+010000	+------------------------+                                          
+	    |  Boot loader		    |	<- Boot sector entry point 0000:7C00    
+001000	+------------------------+                                          
+	    |  Reserved for MBR/BIOS|                                           
+000800	+------------------------+                                          
+	    |  Typically used by MBR|                                           
+000600	+------------------------+                                          
+	    |  BIOS use only	    |                                           
+000000	+------------------------+                                          
+</pre>
+
+
+```rs
+const E820_RAM: u32 = 1;
+```
+RAM memory type.
+e820 is shorthand for the facility by which the BIOS of x86-based computer systems reports the memory map to the operating system or boot loader.
+
+```rs
+pub enum Error {
+    E820Configuration, /// Invalid E820 configuration.
+    HimemStartPastMemEnd, /// Highmem start address is past the guest memory end.
+    HimemStartPastMmioGapStart, /// Highmem start address is past the MMIO gap start.
+    MmioGapPastMemEnd, /// The MMIO gap end is past the guest memory end.
+    MmioGapStartPastMmioGapEnd, /// The MMIO gap start is past the gap end.
+}
+```
+These are different types of errors that can occur during the boot parameter setup.
+
+```rs
+fn add_e820_entry(
+    params: &mut boot_params,
+    addr: u64,
+    size: u64,
+    mem_type: u32,
+) -> result::Result<(), Error>
+```
+This function sets the address, size and memory type in the e820_table of the boot parameters. It will be used in the ```build_bootparams``` function described next.
```

Comment:
Did you describe `e820_table` somewhere? I didn't find it.

Inline code snippets can be in just single ticks instead of three ticks. Like for build_bootparams.

Reply:
![image](https://user-images.githubusercontent.com/53364992/191227690-f7b279a6-24fc-4f8a-baa8-0d1cc25b94bc.png)
I mentioned that e820 contains the memory map. Should I add more ?